core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java

```diff
@@ -1850,6 +1851,41 @@ public SqlNode field(int ordinal, boolean useAlias) {
           needNew && !aliases.containsKey(neededAlias) ? newAliases : aliases);
     }
 
+    boolean isCorrelated(RelNode rel) {
+      if (rel instanceof LogicalFilter && !rel.getVariablesSet().isEmpty()) {
+        List<SqlOperator> correlOperators =
+            Arrays.asList(SqlStdOperatorTable.EXISTS, SqlStdOperatorTable.IN,
+                SqlStdOperatorTable.SCALAR_QUERY);
+
+        List<SqlKind> comparisonOperators =
+            Arrays.asList(SqlKind.NOT, SqlKind.OR,
+                SqlKind.LESS_THAN, SqlKind.GREATER_THAN);
+
+        SqlOperator op = null;
+        RexNode condition = ((LogicalFilter) rel).getCondition();
```

Comment:
We can use the "condition" variable instead of calling the "getCondition" function every time below at 2 places.